libs/wai-utilities/src/Network/Wai/Utilities/Server.hs

```diff
@@ -184,21 +166,38 @@ measureRequests m rtree = withPathTemplate rtree $ \p ->
       requestCounter m p . duration 30 12 m p
 {-# INLINEABLE measureRequests #-}
 
+catchErrors  :: Logger -> OnErrorMetrics -> Middleware
+catchErrors = catchErrors' True
+{-# INLINEABLE catchErrors #-}
+
 -- | Create a middleware that catches exceptions and turns
 -- them into appropriate 'Error' responses, thereby logging
 -- as well as counting server errors (i.e. exceptions that
 -- yield 5xx responses).
-catchErrors :: Logger -> Metrics -> Middleware
-catchErrors l m app req k =
-    app req k `catch` errorResponse
+--
+-- The boolean determines whether response values with
+-- status >= 400 are allowed (servant), or whether we should
+-- not expect and consequently warn about them (wai-route).
+catchErrors' :: Bool -> Logger -> OnErrorMetrics -> Middleware
```

Comment:
This is a case of Boolean blindness I think; I'd prefer to use data here;

```
data NotFoundBehaviour = WarnNotFound | PassThroughNotFound
```

This turns the call site from:

```
catchErrors' True ...
```
to

```
catchErrors' WarnNotFound
```

It's also not clear from the doc which behaviour is given for which; does `true` mean warn or does `true` mean they're okay?

It also means I can understand the arguments by checking the type; `NotFoundBehaviour` has semantic meaning; `Bool` doesn't 😄

Reply:
as discussed, i went for an extra middleware instead.  worked well!